pkg/runs/runsDownload.go

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright contributors to the Galasa project
+ */
+package runs
+
+import (
+	"bufio"
+	"context"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/galasa.dev/cli/pkg/api"
+	galasaErrors "github.com/galasa.dev/cli/pkg/errors"
+	"github.com/galasa.dev/cli/pkg/galasaapi"
+	"github.com/galasa.dev/cli/pkg/utils"
+)
+
+// DownloadArtifacts - performs all the logic to implement the `galasactl runs download` command,
+// but in a unit-testable manner.
+func DownloadArtifacts(
+	runName string,
+	forceDownload bool,
+	fileSystem utils.FileSystem,
+	timeService utils.TimeService,
+	console utils.Console,
+	apiServerUrl string) error {
+
+	var err error = nil
+	var runs []galasaapi.Run
+	var artifactPaths []string
+
+	runs, err = GetRunsFromRestApi(runName, timeService, apiServerUrl)
+	if err == nil {
+		for _, run := range runs {
+			runId := run.GetRunId()
+			artifactPaths, err = GetArtifactPathsFromRestApi(runId, apiServerUrl)
+			if err == nil {
+				for _, artifactPath := range artifactPaths {
+					artifactData, err := GetFileFromRestApi(runId, strings.TrimPrefix(artifactPath, "/"), apiServerUrl)
+					if err != nil {
+						return err
+					} else {
+						err = WriteArtifactToFileSystem(fileSystem, runName, artifactPath, artifactData, forceDownload)
+						if err != nil {
```

Comment:
You can just drop out of the ifs to have error returned at the end...

Reply:
Addressed in https://github.com/galasa-dev/cli/pull/92/commits/ff21f567cdc46463eafc4316afd8bb72ed7bcfdf